Subject: [PATCH] Fix the unlocking by SPACE key event

StatusBar view is handling SPACE key for unlocking,
but QSContainerImpl consumes the event before StatusBar,
so unlocking does not work.

The purpose of QSContainerImpl is to prevent miss touch,
so this patch changes not to consume the event
if phone is not in touch mode.

diff --git a/packages/SystemUI/src/com/android/systemui/qs/QSContainerImpl.java b/packages/SystemUI/src/com/android/systemui/qs/QSContainerImpl.java
index 33b5268..771973d 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/QSContainerImpl.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/QSContainerImpl.java
@@ -61,6 +61,12 @@
 
     @Override
     public boolean performClick() {
+        if (!isInTouchMode()) {
+            // Do not consume Click event if it is not in Touch Mode in order to
+            // handle SPACE key which is mapped as unlocking in StatusBarWindowView.
+            return false;
+        }
+
         // Want to receive clicks so missing QQS tiles doesn't cause collapse, but
         // don't want to do anything with them.
         return true;
